version: '3'

services:
  jupyterhub:
    build: jupyterhub
    image: jupyhub_img
    container_name: jupyhub
    volumes:
      # within the container
      - /var/run/docker.sock:/var/run/docker.sock
      # bind docker volume on host for JupyterHub database (Hub data persistence)
      - jupyterhub_data:/srv/jupyterhub
    environment:
      - DOCKER_JUPYTER_CONTAINER=jupyroot_img
      - DOCKER_NETWORK_NAME=${COMPOSE_PROJECT_NAME}_default
      - HUB_IP=jupyhub
      - HOST
      - OAUTH_CLIENT_ID
      - OAUTH_CLIENT_SECRET
      - OAUTH_CALLBACK_URL
    labels:
      - "traefik.enable=true"
      - "traefik.frontend.rule=Host:${HOST}"
    restart: on-failure

  jupyterlab:
    build: jupyterlab-root-python310
    image: jupyroot_img
    container_name: jupyroot
    network_mode: ${COMPOSE_PROJECT_NAME}_default
    volumes:
      # bind docker volume on host
      - /disk01/jupyter:/home/jovyan/data/user
      - /disk01/cms-open-data:/home/jovyan/data/shared
    command: echo

  reverse-proxy:
    image: traefik:1.7
    container_name: reverse-proxy
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - ./reverse-proxy/traefik.toml:/etc/traefik/traefik.toml
      - /etc/certs:/etc/certs
      - /var/run/docker.sock:/var/run/docker.sock
    restart: on-failure

volumes:
  jupyterhub_data:
